src/func/Microsoft.Health.Fhir.Ingest.Host/IomtConnectorFunctions.cs

```diff
@@ -74,6 +77,15 @@ public IomtConnectorFunctions(ITelemetryLogger logger)
 
                 IDataNormalizationService<EventData, IMeasurement> dataNormalizationService = new MeasurementEventNormalizationService(_logger, template);
                 await dataNormalizationService.ProcessAsync(events, output).ConfigureAwait(false);
+
+                if (normalizationSettings.Value.LogDeviceIngressSizeBytes)
+                {
+                    IEventProcessingMeter meter = new EventProcessingMeter();

```

Comment:
Do we need to create a separate instance of `IEventProcessingMeter` for each invocation of `NormalizeDeviceData`? If not it might make sense to initialize one inside of the constructor. Or, pass in a singleton via dependency injection.

Reply:
Yes I think ideal performance would be to inject IDataNormalizationService<EventData, IMeasurement> and also IEventProcessingMeter but I do not think it is a bottleneck currently, so I did not invest time into doing it.